src/System.Devices.Gpio/System.Devices.Gpio/GPIOPin.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Devices.Gpio
+{
+    public enum GpioPinMode
+    {
+        Input = 0,
+        Output = 1
+    }
+
+    public enum GpioPinValue
+    {
+        Low = 0,
+        High = 1
+    }
+
+    public enum GpioScheme
+    {
+        Board,
+        BCM
+    }
+
+    public class GpioPin
+    {
+        protected GpioDriver Driver;
+
+        public int Number { get; }
+
+        public GpioPinMode Mode
+        {
+            get
+            {
+                return Driver.GetPinMode(Number);
+            }
+            set
+            {
+                Driver.SetPinMode(Number, value);
+            }
+        }
+
+        public GpioPin(GpioDriver driver, GpioScheme numberKind, int number, GpioPinMode mode)
+        {
+            Driver = driver;
+            Number = driver.ConvertPinNumber(number, numberKind, GpioScheme.BCM);
```

Comment:
Does this mean that "Number" is always the BCM pin scheme value?  If so that seems like a bug given the Mode is settable.

Reply:
Yes, it is always BCM number but the GpioPin class also has the GetNumber(GpioScheme kind) method to get the number for a different numbering scheme. Also, the user can specify the pin number using any numbering scheme in the constructor of this class.

An alternative could be having two peroperties: BcmNumber and BoardNumber. But I like more the other approach. Maybe we I should rename the Number property to be BcmNumber instead.